### PR TITLE
Continue reaction sync past the page cap and recover missed replies

### PR DESCRIPTION
**Pull request by pfefferle**
## Proposed changes:
* **Resume pagination across runs.** Reaction sync previously advanced its watermark after five pages (250 notifications) per run, permanently stranding every older reply and reaction on busy accounts. The continuation cursor is now persisted per stream and resumed on the next run, with a one-time replay of the completed range so nested replies processed in an earlier chunk than their parent still thread correctly. Repeated continuation failures (3×) discard the cursor and restart from the newest page.
* **Daily rolling reply audit.** A new daily cron checks a small batch of published posts (default 5, filterable via `atmosphere_reply_backfill_batch_size`, `0` disables) against Bluesky's public `getPostThread` and imports replies that are missing locally — recovering replies that were already stranded before this fix. Never-checked posts go first; afterwards the least-recently checked posts rotate through.
* **On-demand recovery via WP-CLI.** `wp atmosphere replies backfill <post-id>` scans one post's thread immediately and reports found / imported / already present / skipped / pending counts.
* **Shared lease lock.** The hourly sync and both backfill paths coordinate through a renewable cross-process lease lock (new reusable `Atmosphere\Lock` class: atomic options-table acquire, owner token, CAS steal of expired rows, renewal throttled to fire only when less than half the TTL remains) so they never run concurrently. `OAuth\Client`'s older lock can migrate to it in a follow-up.
* **Moderation-safe dedup.** The source-URI dedup lookups now include spammed/trashed comments, so a reply the admin moderated away is not re-imported by a later audit.
* Thread imports are capped at 500 reply nodes per run, and the daily audit is scheduled offset by 30 minutes so its due timestamps never collide with the hourly sync's.

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* `npm run env-start`, connect the plugin to a Bluesky account, and publish a post.
* Reply to the post on Bluesky (add a nested reply to your reply as well), then run `wp-env run cli wp atmosphere replies backfill <post-id>`: both replies should appear as threaded comments and the summary should report `imported: 2`. Run it again — `already present: 2`, no duplicates.
* Mark one imported reply as spam in wp-admin and re-run the command: the spammed reply must **not** come back (`already present` still counts it).
* For the overflow fix: `npm run env-test -- --filter=paginate` covers watermark, continuation, replay, and the page-budget boundary cases; `npm run env-test -- --filter='backfill|replies|lock'` covers the audit, CLI command, and lock semantics.
* Full suite: `composer lint` and `npm run env-test` (929 tests passing).

### CODE REVIEW

File: includes/class-reaction-sync.php [hunk 1]
```diff
@@ -1161,9 +1940,56 @@ private static function find_comment_by_source_id( string $uri ): int|false {
...
+
+		foreach ( $comments as $comment ) {
+			$uri = (string) \get_comment_meta( (int) $comment->comment_ID, self::META_SOURCE_ID, true );
+
+			if ( '' !== $uri ) {
+				$existing[ $uri ] = true;
+			}
+		}
+
+		return $existing;
```
[the remark this answers is not in the document]
Comment (pfefferle) [reply includes/class-reaction-sync.php:2037]: This is already one batched query, not N+1: `get_comments()` defaults `update_comment_meta_cache` to `true`, which queues all found comment IDs via `wp_lazyload_comment_meta()` — the first `get_comment_meta()` call in the loop fires a single batched meta query for every queued ID, and the remaining iterations are cache hits (see `WP_Comment_Query::get_comment_ids()` → line 485 in core). Total cost is one comment query plus one meta query regardless of thread size, which is exactly the "prime in one query" shape suggested here.

**Review by jeherve — COMMENTED** [review]
A couple of behavior questions from the review

1. resolve_author() when getProfile fails (class-reaction-sync.php:1683)

I noticed resolve_author() returns array() on a getProfile WP_Error, and it's the only is_wp_error branch in the file that returns without a debug_log. The callers (process_reply, process_subject_reaction, process_own_record) then insert the comment anyway with an empty name/handle/avatar.

The reason I'm flagging this one over a normal transient blip: the success transient is only written after getProfile succeeds (:1693), so a failure isn't cached, but the comment still gets imported. Once the watermark moves past it and source_id dedup kicks in, it never gets revisited. So a brief rate-limit or 5xx ends up baked in as a permanent comment with a missing avatar, and for own-record reactions an empty handle, which gives a broken comment_author_url. The daily reply-backfill re-audits replies but not likes/reposts, so those don't self-heal either.

I'm wondering if importing-with-empty-profile is the degradation you actually want here, or whether a getProfile failure should defer that reaction so a later run retries it. And either way, could we add a debug_log on that branch? It's the one failure in the file that leaves no trace today, so right now we'd have no way to spot it in the field. What do you think?

2. DEDUP_COMMENT_STATUSES and parent resolution (class-reaction-sync.php:150, used at :1327)

The widened status set (adding spam/trash/post-trashed) makes total sense for the dedup lookup, and the docblock explains it well; without it the backfill re-imports a reply the admin moderated away on every audit. No concern there.

The bit I want to double-check is intent. find_comment_by_source_id() is also the parent-resolution lookup in process_reply() (:1327), so it inherits the wider set too. Before this change, a reply whose parent had been spammed or trashed would fail to resolve its parent and get dropped as an orphan at :1360. After it, that parent resolves, so the child gets imported and attached to a comment the admin had removed; and if the child passes its own moderation, it shows up as a visible sub-thread hanging off a suppressed parent.

Is that intended? The :1339 rationale is all about orphan and out-of-order handling and doesn't mention moderated parents, so it reads to me like an incidental side effect of one lookup doing two jobs. If we'd rather not resurface under a moderated parent, one option is to keep dedup on the wide set but resolve parents only among approve/hold with a second, narrower lookup for the :1327 call. Curious what you were picturing here.

**Comment by pfefferle** [comment]
Both good catches — thanks for the careful read. Addressed in 0a1c2ac:

**1. `resolve_author()` failures** — added the missing `debug_log` on that branch (you're right that it was the only untraced failure in the file). One nuance: for notification-stream and thread-backfill reactions, `insert_reaction()` already falls back to the payload's own `author.handle`, so a getProfile blip there costs only the display name and avatar — not the author URL. The genuinely broken case was own-record reactions, where the handle itself came from getProfile; those now fall back to the locally stored identity handle, so the author link can't break at all. On defer-vs-import: I kept import-with-degraded-profile deliberately — a permanently deleted account's getProfile fails forever, so deferral would mean those reactions never import, and the reaction content matters more than its avatar. With the fallbacks, the permanent loss is displayName/avatar only, now with a breadcrumb.

**2. Moderated parents** — you read it exactly right: the widening was meant for dedup and leaked into parent resolution as a side effect of one lookup doing two jobs. The orphan rationale at the call site even lists "removed from our moderation queue" as a drop case, so this was a regression of documented intent. Fixed per your suggestion: `find_comment_by_source_id()` takes a status list now — dedup keeps the wide set, parent resolution passes `approve`/`hold` only — with a regression test (spammed parent → incoming child drops as an orphan).

**Review by jeherve — APPROVED** [review]
